src/main/java/net/irisshaders/iris/shaderpack/parsing/BooleanParser.java

```diff
@@ -0,0 +1,150 @@
+		try {
+			String option = "";
+			Stack<OPERATION> operationStack = new Stack<>();
+			Stack<Boolean> valueStack = new Stack<>();
+			for (int i = 0; i < expression.length(); i++) {
+				char c = expression.charAt(i);
+				switch (c) {
+					case '!' -> operationStack.push(OPERATION.NOT);
+					case '&' -> {
+						// add value first, because this checks for preceding NOTs
+						if (!option.isEmpty()) {
+							valueStack.push(processValue(option, valueLookup, operationStack));
+							option = "";
+						}
+						// AND operators have priority, so add a bracket if it's the first AND
+						if (operationStack.isEmpty() || !operationStack.peek().equals(OPERATION.AND)) {
+							operationStack.push(OPERATION.OPEN);
+						}
+						i++;
+						operationStack.push(OPERATION.AND);
+					}
+					case '|' -> {
+						// add value first, because this checks for preceding NOTs
+						if (!option.isEmpty()) {
+							valueStack.push(processValue(option, valueLookup, operationStack));
+							option = "";
+						}
+						// if there was an AND before, that needs to be evaluated because it takes priority
+						if (!operationStack.isEmpty() && operationStack.peek().equals(OPERATION.AND)) {
+							evaluate(operationStack, valueStack, true);
+						}
+						i++;
+						operationStack.push(OPERATION.OR);
+					}
+					case '(' -> operationStack.push(OPERATION.OPEN);
+					case ')' -> {
+						// add value first, because this checks for preceding NOTs
+						if (!option.isEmpty()) {
+							valueStack.push(processValue(option, valueLookup, operationStack));
+							option = "";
+						}
+						// if there was an AND before, that needs to be evaluated because it added its own bracket
+						if (!operationStack.isEmpty() && operationStack.peek().equals(OPERATION.AND)) {
+							evaluate(operationStack, valueStack, true);
+						}
+						evaluate(operationStack, valueStack, true);
+					}
+					case ' ' -> {}
+					default -> option += c;
+				}
+			}
+			if (!option.isEmpty()) {
+				valueStack.push(processValue(option, valueLookup, operationStack));
+			}
+			evaluate(operationStack, valueStack, false);
+			boolean result = valueStack.pop();
+			if (!valueStack.isEmpty() || !operationStack.isEmpty()) {
+				Iris.logger.warn(
+					"Failed to parse the following boolean operation correctly, stacks not empty, defaulting to true!: '{}'",
+					expression);
+				return true;
+			}
+			return result;
+		} catch (EmptyStackException emptyStackException) {
+			Iris.logger.warn(
+				"Failed to parse the following boolean operation correctly, stacks empty when it shouldn't, defaulting to true!: '{}'",
+				expression);
+			return true;
+		}
+	}
+
+	/**
+	 * gets the value for the given string and negates it if there is a NOT in the operationStack
+	 */
+	private static boolean processValue(String value, OptionValues valueLookup, Stack<OPERATION> operationStack) {
+		boolean booleanValue = switch (value) {
+			case "true", "1" -> true;
+			case "false" -> false;
```

Comment:
does a `0` literal also exist?

Reply:
don't think so, the "1" is just there, because the preprocessor replaces stuff that is defined with a 1

Reply:
I can add a "0" aswell though